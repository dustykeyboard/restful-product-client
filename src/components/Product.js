import React from 'react'
import PropTypes from 'prop-types'
import '../styles/product.css'

const Product = ({ id, product_name, description, product_image, price }) => (
  <div className="product">
    <img className="product_image" src={product_image} alt="{product_name}" width={340} height={340} />
    <h2 className="product_name">{product_name}</h2>
    <p className="product_description">{description}</p>
    <p className="product_price">{price}</p>
  </div>
)

Product.propTypes = {
  id: PropTypes.number.isRequired,
  product_name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  product_image: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
}

export default Product
