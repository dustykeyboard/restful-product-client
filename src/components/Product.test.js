import React from 'react'
import renderer from 'react-test-renderer'
import Product from './Product'

it('renders correctly', () => {
  const renderedComponent = renderer
    .create(
      <Product
        id={123}
        product_name="Product Name"
        product_image="product.png"
        description="Description"
        price="$9.99"
      />,
    )
    .toJSON()

  expect(renderedComponent).toMatchSnapshot()
})
