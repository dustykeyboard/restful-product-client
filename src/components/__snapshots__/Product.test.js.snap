// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="product"
>
  <img
    alt="{product_name}"
    className="product_image"
    height={340}
    src="product.png"
    width={340}
  />
  <h2
    className="product_name"
  >
    Product Name
  </h2>
  <p
    className="product_description"
  >
    Description
  </p>
  <p
    className="product_price"
  >
    $9.99
  </p>
</div>
`;
